packages/editor/src/components/media-placeholder/index.js

```diff
@@ -193,43 +207,74 @@ export class MediaPlaceholder extends Component {
 			}
 		}
 
+		const dropZone = (
+			<DropZone
+				onFilesDrop={ this.onFilesUpload }
+				onHTMLDrop={ onHTMLDrop }
+			/>
+		);
+
+		if ( dropZoneUIOnly ) {
+			return (
+				<MediaUploadCheck>
+					{ dropZone }
+				</MediaUploadCheck>
+			);
+		}
+
 		return (
 			<Placeholder
 				icon={ icon }
 				label={ title }
 				instructions={ instructions }
-				className={ classnames( 'editor-media-placeholder', className ) }
+				className={
+					classnames(
+						'editor-media-placeholder',
+						className,
+						{ 'is-appender': isAppender }
+					)
+				}
 				notices={ notices }
 			>
 				<MediaUploadCheck>
-					<DropZone
-						onFilesDrop={ this.onFilesUpload }
-						onHTMLDrop={ onHTMLDrop }
-					/>
+					{ dropZone }
 					<FormFileUpload
 						isLarge
-						className="editor-media-placeholder__button"
+						className={ classnames(
+							'editor-media-placeholder__button',
+							'editor-media-placeholder__upload-button'
+						) }
 						onChange={ this.onUpload }
 						accept={ accept }
 						multiple={ multiple }
 					>
 						{ __( 'Upload' ) }
 					</FormFileUpload>
 					<MediaUpload
+						addToGallery={ addToGallery }
```

Comment:
Is there a use case when we'd disable `addToGallery`? I'd expect folks to always need to opt-in to this behavior when using the `MediaPlaceholder`.

It also seems to overlap with `isAppender`, although one is for styling and the other behavior.

Reply:
> Is there a use case when we'd disable `addToGallery`? I'd expect folks to always need to opt-in to this behavior when using the `MediaPlaceholder`.

When the gallery/value was empty previously we went to the create gallery flow instead of the add to gallery one. If we default to addToGallery we would be applying a behavior change and go by default to the add flow instead of the create flow.

> It also seems to overlap with `isAppender`, although one is for styling and the other behavior.

Yes, I preferred to have a flag for behavior and another one for the style. So we allow more customization.

Thank you for the review @sirreal 👍